faillint/faillint.go

```diff
@@ -5,134 +5,176 @@ package faillint
 import (
 	"fmt"
 	"go/ast"
+	"go/token"
+	"regexp"
 	"strconv"
 	"strings"
+	"unicode"
 
 	"golang.org/x/tools/go/analysis"
 )
 
+var (
+	pathRegexp = regexp.MustCompile(`(?P<import>[\w/.-]+[\w])(\.?{(?P<functions>[\w-,]+)}|)(=(?P<suggestion>[\w/.-]+[\w](\.?{[\w-,]+}|))|)`)
+)
+
 type faillint struct {
 	paths       string // -paths flag
 	ignoretests bool   // -ignore-tests flag
 }
 
-// Analyzer global instance of the linter (if possible use NewAnalyzer)
-var Analyzer = NewAnalyzer()
-
-// NewAnalyzer create a faillint analyzer
-func NewAnalyzer() *analysis.Analyzer {
+// New create a faillint analyzer.
+func New() *analysis.Analyzer {
```

Comment:
I like NewAnalyzer better, would be happy if you can revert the naming.

Reply:
Why? To me it's like including the type in variable name... (: `mInt` `someVarString` .. we used to do that 20 years ago when there were no modern IDEs.. now I think it's just a noise. You can check return type to learn about that it's analyser ((: Wdyt? What would be the argument to have Analyser in name?

Reply:
Just using `New()` doesn't explain immediately what it is returning. However `NewAnalyzer()` means exactly what it is. You're creating a new analzyer. You can also see some examples in the stdlib. For example from the `bufio` package:

NewReader()
NewBuffer()
NewWriter

Or from the `flag` package:

NewFlagSet()

There are many examples like this. You could argue that `errors` has a `New()` function, but that's a little bit different than these.

There it would be repetitive to say `errors.NewError()` and it's used in every single return expression as well, such as `return nil, errors.New()`. So keeping it short is important.

But for faillint, or any other package that provide a certain functionality, I would like to see the name of _what_ it is creating in the `New...` name.

Reply:
Fair, but currently we have only one constructor, so if we would follow [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) rule, `New` is just enough. (: 

Let me know if that convinces you. If not, I will disagree but revert back and commit :smile:

Reply:
You're hard to convince I have to admit it :) 

So yeah I still thing `NewAnalyzer()` is better than `New()` for this _package_. This is used only in one place, in our cmd/main.go file, but I want it to be descriptive. I don't think `New()` is descriptive enough. Thank you for understanding, I appreciate your co-operation .

Reply:
Cool, Reverting :+1:

Reply:
Done.